src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionSettings.cs

```diff
@@ -131,55 +136,59 @@ public HttpConnectionSettings CloneAndNormalize()
             return settings;
         }
 
-        private static bool AllowHttp2
+        // Default to allowing HTTP/2, but enable that to be overridden by an
+        // AppContext switch, or by an environment variable being set to false/0.
+        private static bool AllowHttp2 => RuntimeSettingParser.QueryRuntimeSettingSwitch(
+            "System.Net.Http.SocketsHttpHandler.Http2Support",
+            "DOTNET_SYSTEM_NET_HTTP_SOCKETSHTTPHANDLER_HTTP2SUPPORT",
+            true);
+
+        // Default to allowing draft HTTP/3, but enable that to be overridden
+        // by an AppContext switch, or by an environment variable being set to false/0.
+        private static bool AllowDraftHttp3 => RuntimeSettingParser.QueryRuntimeSettingSwitch(
+            "System.Net.SocketsHttpHandler.Http3DraftSupport",
+            "DOTNET_SYSTEM_NET_HTTP_SOCKETSHTTPHANDLER_HTTP3DRAFTSUPPORT",
+            true);
+
+        // Switch to disable the HTTP/2 dynamic window scaling algorithm. Enabled by default.
+        private static bool DisableDynamicHttp2WindowSizing => RuntimeSettingParser.QueryRuntimeSettingSwitch(
+            "System.Net.SocketsHttpHandler.Http2FlowControl.DisableDynamicWindowSizing",
+            "DOTNET_SYSTEM_NET_HTTP_SOCKETSHTTPHANDLER_HTTP2FLOWCONTROL_DISABLEDYNAMICWINDOWSIZING",
+            false);
+
+        // The maximum size of the HTTP/2 stream receive window. Defaults to 16 MB.
+        private static int MaxHttp2StreamWindowSize
```

Comment:
No consensus yet, see https://github.com/dotnet/runtime/pull/54755#issuecomment-869949514 and https://github.com/dotnet/runtime/pull/54755#issuecomment-869978268. Curious about your opinion.

Reply:
I'm not a fan of "what-if" configurability unless we have a very strong use-case for it so that's why I was asking. But the consensus was to keep them, so keep them.